examples/basic/__tests__/App.test.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react-native';
+import App from '../App';
+
+/**
+ * A good place to start is having a tests that your component renders correctly.
+ */
+test('renders correctly', () => {
+  // Idiom: no need to capture render output, as we will use `screen` for queries.
+  render(<App />);
+
+  // Idiom: `getByXxx` is a predicate by itself, but we will use it with `expect().toBeTruthy()`
+  // to clarify our intent.
+  expect(screen.getByText('Sign in to Example App')).toBeTruthy();
+});
+
+/**
+ * Hint: It's best when your tests are similar to what a manual test scenarions would look like,
+ * i.e. a series of actions taken by the user, followed by a series of assertions verified from
+ * his point of view.
+ */
+test('User can sign in successully with correct credentials', async () => {
+  // Idiom: no need to capture render output, as we will use `screen` for queries.
+  render(<App />);
+
+  // Idiom: `getByXxx` is a predicate by itself, but we will use it with `expect().toBeTruthy()`
+  // Note: `.toBeTruthy()` is the preferred matcher for checking that elements are present.
+  expect(screen.getByText('Sign in to Example App')).toBeTruthy();
+  expect(screen.getByText('Username')).toBeTruthy();
+  expect(screen.getByText('Password')).toBeTruthy();
+
+  // Hint: we can use `getByLabelText` to find our text inputs in accessibility-friendly way.
+  fireEvent.changeText(screen.getByLabelText('Username'), 'admin');
+  fireEvent.changeText(screen.getByLabelText('Password'), 'admin1');
+
+  // Hint: we can use `getByText` to find our button by its text.
+  fireEvent.press(screen.getByText('Sign In'));
+
+  // Idiom: since pressing button triggers async operation we need to use `findBy` query to wait
+  // for the action to complete.
+  // Hint: subsequent queries do not need to use `findBy`, because they are used after the async action
+  // already finished
+  expect(await screen.findByText('Welcome admin!')).toBeTruthy();
```

Comment:
There's no need to expect everything, right? This reads better as a scenario to me:
```suggestion
  await screen.findByText('Welcome admin!'));
```

Reply:
Yeah thats a good question, I'used that in a similar pattern as we suggest using `expect(getByText()).toBeTruthy()` to clarify the intent, despite the fact that simple `getByText` would do the same.